### ISSUE TITLE
Add Elasticsearch/Loki bulk index as output formats

### ISSUE THREAD

**Issue by lenny-ts** [lenny-ts/caddy-analyzer#74]
### Problem

caddy-analyzer can read from files, Docker, K8s, and journalctl, but analysis results can only be printed to stdout or written to local files. Teams running Elasticsearch/OpenSearch or Grafana Loki cannot index caddy-analyzer output into their existing observability infrastructure.

### Proposed Solution

Add two new output formats:

**1. Elasticsearch bulk index** (`-f elasticsearch`):

```bash
caddy-analyze -f elasticsearch --es-url http://localhost:9200 --es-index caddy-analysis access.log
```

Uses the `_bulk` API to index the analysis report as a single document. Include timestamp, RPS, status breakdown, detection results, and per-IP summaries.

**2. Grafana Loki push** (`-f loki`):

```bash
caddy-analyze -f loki --loki-url http://localhost:3100 --loki-labels job=caddy-analyzer access.log
```

Uses the `/loki/api/v1/push` API to send log entries with labels (`job`, `host`, `severity`).

### Implementation Notes

- Add an `Exporter` interface in `pkg/output/` analogous to the `Enricher` interface in `pkg/enrich/`
- Implement `ElasticsearchExporter` and `LokiExporter`
- For Elasticsearch: batch documents with `refresh=false`, flush on report completion
- For Loki: send entries with structured labels matching the report dimensions
- Both should be async (buffer in background goroutine) to avoid blocking the analysis loop
- Add retry with exponential backoff on HTTP failures

### Impact

Critical for teams running ELK or Loki who want to index Caddy analysis results alongside other logs. Without this, caddy-analyzer is a standalone CLI tool that cannot integrate with centralized observability.
